File: WebApplication/src/modules/application/application-one/models/application-one.dto.ts
import { AuthorityDto } from "../../../nomenclature/dtos/authority.dto";
import { ApplicationAuthorityDto } from "../../common/models/application-authority.dto";
import { ApplicationFileDto } from "../../common/models/application-file.dto";
import { CommitDto } from "../../common/models/commit.dto";
import { ApplicationOneType } from "../enums/application-one-type.enum";
import { ApplicantDto } from "./applicant.dto";
import { ApplicationOneAuthorityDto } from "./application-one-authority.dto";
import { ApplicationOneBasicDto } from "./application-one-basic.dto";
import { ApplicationOneDamageDto } from "./application-one-damage.dto";
import { ApplicationOneLegalEntityDto } from "./application-one-legal-entity.dto";
import { ApplicationOneThreatDto } from "./application-one-threat.dto";

export class ApplicationOneDto extends CommitDto {
  id: number;
  registerNumber: string;
  applicationOneType: ApplicationOneType;

  applicationOneAuthorities: ApplicationAuthorityDto[] = [];

  applicantId: number;
  applicant: ApplicantDto;

  applicationOneBasic: ApplicationOneBasicDto;
  applicationOneLegalEntity: ApplicationOneLegalEntityDto;
  applicationOneThreat: ApplicationOneThreatDto;
  applicationOneDamage: ApplicationOneDamageDto;

  applicationOneFiles: ApplicationFileDto[] = [];
  hasApplicationTwo: boolean;
  applicationTwoRootId: number;

  constructor() {
    super()

    this.applicant = new ApplicantDto();
    this.applicationOneBasic = new ApplicationOneBasicDto();
  }

  clearAll() {
    this.clearOptionals();

    this.applicant.clearAll();
  }

  clearOptionals() {
    this.applicationOneLegalEntity = new ApplicationOneLegalEntityDto();
    this.applicationOneThreat = new ApplicationOneThreatDto();
    this.applicationOneDamage = new ApplicationOneDamageDto();
  }
}

File: Zopoesht.Application/Applications/ApplicationsOne/Dtos/ApplicationOneDto.cs
using Zopoesht.Application.Applications.Common.Dtos;
using Zopoesht.Data.Applications.ApplicationOne.Enums;
using Zopoesht.Infrastructure.DomainValidation;
using Zopoesht.Infrastructure.DomainValidation.Enums;

namespace Zopoesht.Application.Applications.ApplicationsOne.Dtos
{
    public class ApplicationOneDto : ApplicationCommitDto
    {
        public string RegisterNumber { get; set; }
        public bool HasApplicationTwo { get; set; }
        public int? ApplicationTwoRootId { get; set; }

        public ApplicationOneType ApplicationOneType { get; set; }

        public List<ApplicationAuthorityDto> ApplicationOneAuthorities { get; set; } 

        public int? ApplicantId { get; set; }
        public ApplicantDto Applicant { get; set; }

        public ApplicationOneBasicDto ApplicationOneBasic { get; set; }

        public ApplicationOneLegalEntityDto ApplicationOneLegalEntity { get; set; }

        public ApplicationOneThreatDto ApplicationOneThreat { get; set; }

        public ApplicationOneDamageDto ApplicationOneDamage { get; set; }

        public List<ApplicationFileDto> ApplicationOneFiles { get; set; } = new List<ApplicationFileDto>();

        public override void ValidateProperties(DomainValidationService validationService)
        {
            if (!Enum.IsDefined(typeof(ApplicationOneType), ApplicationOneType))
            {
                validationService.ThrowErrorMessage(ValidationErrorCode.Validation_InvalidFormat);
            }

            Applicant.ValidateProperties(validationService);

            ApplicationOneBasic.ValidateProperties(validationService);


            if (ApplicationOneType == ApplicationOneType.ReportedDamage)
            {
                ApplicationOneLegalEntity.ValidateProperties(validationService);
                ApplicationOneLegalEntity.ValidateApplicantViolations(Applicant.ApplicantType, validationService);
            }

            if (ApplicationOneType == ApplicationOneType.Threat)
            {
                ApplicationOneThreat.ValidateProperties(validationService);
            }

            if (ApplicationOneType == ApplicationOneType.Damage)
            {
                ApplicationOneDamage.ValidateProperties(validationService);
            }
        }
    }
}
